.github/workflows/static_checks.yaml

```diff
@@ -0,0 +1,23 @@
+name: Static Checks
+on:
+    pull_request: {}
+    push: {}
+jobs:
+    go_lint:
```

Comment:
Wondering if we could use https://github.com/golangci/golangci-lint directly, so that we don't have to take the hassle of maintaining all these tools induvidually :thinking:

Reply:
Ah, that makes sense! I think we should directly go ahead with the action it provides - https://golangci-lint.run/usage/install/#ci-installation